### PR TITLE
don't let users include certain words during user rename

### PR DESCRIPTION
**Pull request by NatalieEO**
https://ucsc-cgl.atlassian.net/browse/SEAB-1800

### CODE REVIEW

File: dockstore-webservice/src/test/java/io/dockstore/webservice/resources/UserResourceTest.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+import io.dockstore.webservice.CustomWebApplicationException;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class UserResourceTest {
+
+    @Test
```
Comment (coverbeck) [inline dockstore-webservice/src/test/java/io/dockstore/webservice/resources/UserResourceTest.java:12]: 👍 for unit test

**Review by denis-yuen — APPROVED** [review]

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java [hunk 2]
```diff
@@ -264,6 +267,16 @@ public User changeUsername(@ApiParam(hidden = true) @Parameter(hidden = true, na
         return userDAO.findById(user.getId());
     }
 
+    public static void restrictUsername(String username) {
+        Pattern pattern = Pattern.compile("(?i)^(?!.*(dockstore|admin|curator|system|manager)).*$");
```
[the remark this answers is not in the document]
Comment (NatalieEO) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java]: After some discussion, not addressing third point. Okay for now.

**Review by garyluu — APPROVED** [review]
Seems fine, something to consider in the future is to also ban offensive words.

File: dockstore-webservice/src/test/java/io/dockstore/webservice/resources/UserResourceTest.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    public void usernameRestrictionsTest() {
+        List<String> invalidUsernames = new ArrayList<>(List.of("dockstore", "DOCKSTORE", "thisIsDoCkStOrE", "admin", "aaaaadminbbb", "curator",
```
Comment (coverbeck) [inline dockstore-webservice/src/test/java/io/dockstore/webservice/resources/UserResourceTest.java:14]: Ideally you would have some valid usernames as well, to make sure your code just doesn't always reject everything. Not that it does, and it's pretty straightforward. Just for future reference.

**Review by coverbeck — APPROVED** [review]

**Comment by denis-yuen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- dockstore-webservice/src/test/java/io/dockstore/webservice/resources/UserResourceTest.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/dockstore/dockstore/pullRequest?prid=6271593&bid=20415219)

**Comment by denis-yuen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- dockstore-webservice/src/test/java/io/dockstore/webservice/resources/UserResourceTest.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/dockstore/dockstore/pullRequest?prid=6271593&bid=20415219)

**Comment by denis-yuen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- dockstore-webservice/src/test/java/io/dockstore/webservice/resources/UserResourceTest.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/dockstore/dockstore/pullRequest?prid=6271593&bid=20415219)

**Pull request merged**